PublicFolders/ValidateEXOMePf.ps1

```diff
@@ -0,0 +1,490 @@
+        if ($null -ne $DistinguishedName -and $DistinguishedName.contains("OU=Soft Deleted Objects")) {
+            $Fix= "FIX -->Please follow the following article https://learn.microsoft.com/en-us/exchange/collaboration-exo/public-folders/recover-deleted-public-folder-mailbox
+            to validate if the content public folder mailbox:$PublicFolderMbxGuid was soft-deleted then restore it back"
+            $Issue="Public folder content mailbox $PublicFolderMbxGuid is soft deleted which is a blocker for receiving emails over the mail public folder"
+            WriteToScreenAndLog -Issue $Issue -Fix $Fix
+            AskForFeedback
+            QuitEXOSession
+        } else {
+            #PfMbx is hard deleted
+            $Fix= "FIX -->Please follow the below steps to create a new public folder using below steps:
+                            1-Follow the following article https://aka.ms/disablempf to mail disable the affected public folder
+                            2-Create public folder using following article https://aka.ms/newpf with same name $($MailPublicFolder.Name)
+                            3-Follow the following article https://aka.ms/EnableMPF to mail enable the newly created public folder with the same email address $($MailPublicFolder.PrimarySmtpAddress)"
+            $Issue="Public folder mailbox $DistinguishedName is hard deleted(Purged) which is a blocker for receiving emails over the public folder"
+            WriteToScreenAndLog -Issue $Issue -Fix $Fix
+            AskForFeedback
+            QuitEXOSession
+        }
+    }
+}
+function CheckMePfHealth {
+    param([PSCustomObject]$MailPublicFolder)
+    try {
+        $LegacyExchangeDN=$MailPublicFolder.LegacyExchangeDN
+        $count=$LegacyExchangeDN.split("=").count
+        $PublicFolder=Get-PublicFolder $LegacyExchangeDN.split("=")[$count-1] -ErrorAction stop
+        if ($PublicFolder.MailEnabled -eq $true) {
+            $HasValue=$PublicFolder.MailRecipientGuid.Guid
+            if ($null -eq $HasValue -or $HasValue -eq "00000000-0000-0000-0000-000000000000") {
+                #MailRecipientGuid is null give action & quit
+                $Fix= "FIX --> Please follow the following article https://aka.ms/disablempf to mail disable the affected public folder then follow the following article https://aka.ms/EnableMPF to mail enable the affected public folder to generate a GUID over MailRecipientGuid parameter, validate MailRecipientGuid parameter has a GUID using https://aka.ms/getpf article."
+                $Issue="Mail-enabled public folder $($PublicFolderInfo.PublicFolder.Identity) is unhealthy e.g MailRecipientGuid parameter is found empty/null which is a blocker for receiving emails over the public folder"
+                WriteToScreenAndLog -Issue $Issue -Fix $Fix
+                AskForFeedback
+                QuitEXOSession
```

Comment:
This function ends up quitting the powershell session in the middle of the script. We should not be doing this. If you can't continue because of what we did, then you need to properly handle the exit of the script as well.

Reply:
This function is ending the script as there's no need to continue checking other functions since the cause was obvious which as orphaned mail-enabled public folder with public folder set to True while the Guid is null or Zeroes.

Reply:
The problem is the way the code is written you will continue to those other functions within the script checking other things and running other cmdlets because you do not properly exit here. If the intent is to exit and be done checking, then truly exit don't just close the session.

Reply:
I assume replacing the break command with exit under QuitEXOSession function shall be fulfilling same request to exit and kill the session, should that be sufficient?

Reply:
No, a `break` is an exit to the current control block, not the same thing as exiting the script.

Reply:
Added Exit after QuitEXOSession function to exit the script.